authhelper.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Pallinder/go-randomdata"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/muir/nfigure"
+	"github.com/pkg/browser"
+)
+
+// AuthHelperOutput is written to STDOUT as JSON
+type AuthHelperOutput struct {
+	ExpiresAt          string `json:"expire"`   // RFC3339 format
+	PasswordToken      string `json:"password"` // A JWT
+	Username           string `json:"username"`
+	Email              string `json:"email,omitempty"`
+	ModelVersionNumber int    `json:"modelVersionNumber"`
+}
+
+func main() {
+	main2(os.Stdout)
+}
+
+func getConfig() (config struct {
+	BaseURL   string   `flag:"baseURL" default:"https://portal.singlestore.com/engine-sso" help:"override the URL passed to the browser"`
+	Email     string   `flag:"email" validate:"omitempty,validate" help:"users SSO email address, if known"`
+	ClusterID []string `flag:"cluster-id,split=comma" help:"comma-separated list of specific clusters to access"`
+	Databases []string `flag:"databases,split=comma" help:"comma-separated list of specific databases to access"`
```

Comment:
Are these and BaseURL for some future use? (these are not currently passed from the driver)

Reply:
BaseURL is used for FE development.
ClusterID & Databases are for future use.